### PR TITLE
ZOOKEEPER-4461: Use logback in zookeeper-contrib project

### PR DESCRIPTION
**Pull request by rahulrane50**
### Description
Cherry picking second commit for log4j to logback : [PR](https://github.com/apache/zookeeper/commit/04c7ad115e298387e4a75dfe14013f6c6aa68a6d)

Cherry picked fix for flaky ctest : [PR](https://github.com/apache/zookeeper/pull/1827)

### Tests
CI and end-to-end tests pipeline

Local zk server testing.

### Changes that Break Backward Compatibility (Optional)
NA

### Documentation (Optional)
NA

### CODE REVIEW

**Comment by rahulrane50** [comment]
@desaikomal

File: zookeeper-client/zookeeper-client-c/tests/TestOperations.cc [hunk 1]
```diff
@@ -126,7 +126,7 @@ class Zookeeper_operations : public CPPUNIT_NS::TestFixture
         zh=zookeeper_init("localhost:2121",watcher,10000,TEST_CLIENT_ID,0,0);
         CPPUNIT_ASSERT(zh!=0);
         // simulate connected state
-        forceConnected(zh);
+        forceConnected(zh, &timeMock.tv);
```
Comment (desaikomal) [inline zookeeper-client/zookeeper-client-c/tests/TestOperations.cc:129]: this is not related to logback change.  shouldn't this be different CL# which fixes just test issues?

Comment (rahulrane50) [reply zookeeper-client/zookeeper-client-c/tests/TestOperations.cc:129]: yeah ideally i should have separated out these but since it was small fix i added it in same PR. Hope it's okay or i can create a new PR for that :) (updated description of PR as well.)

Comment (desaikomal) [reply zookeeper-client/zookeeper-client-c/tests/TestOperations.cc:129]: With this change, everything is completely tested? or do you have more changes?

Comment (rahulrane50) [reply zookeeper-client/zookeeper-client-c/tests/TestOperations.cc:129]: Nope yes basically CI should be passed.

File: zookeeper-client/zookeeper-client-c/tests/ZKMocks.cc [hunk 2]
```diff
@@ -534,8 +534,13 @@ void forceConnected(zhandle_t* zh){
...
     zh->input_buffer=0;
-    gettimeofday(&zh->last_recv,0);
-    gettimeofday(&zh->last_send,0);
+    if (last_recv_send) {
+        zh->last_recv = *last_recv_send;
+        zh->last_send = *last_recv_send;
+    } else {
+        gettimeofday(&zh->last_recv,0);
+        gettimeofday(&zh->last_send,0);
+    }
```
Comment (qqu0127) [inline zookeeper-client/zookeeper-client-c/tests/ZKMocks.cc:543]: For my own knowledge, what is this change, is this related to logback?

Comment (rahulrane50) [reply zookeeper-client/zookeeper-client-c/tests/ZKMocks.cc:543]: Nope it's not related to logback but fixing one flaky c test in CI. My bad i should have separated out these changes will keep that in mind from next time :)

**Review by mgao0 — APPROVED** [review]
lgtm
